objects/src/accounts/account_id.rs

```diff
@@ -123,6 +136,133 @@ impl AccountId {
+            let mut init_seed = init_seed;
+            init_seed[0] += count as u8;
+            spawn(move || {
+                Self::get_account_seed_inner(
+                    send,
+                    stop,
+                    init_seed,
+                    account_type,
+                    on_chain,
+                    code_root,
+                    storage_root,
+                )
+            });
+        }
+
+        #[allow(unused_variables)]
+        let (digest, seed) = recv.recv().unwrap();
+        let _ = stop.set(true);
+
+        #[cfg(feature = "log")]
+        {
+            log::info!(
+                "Found account seed [pow={}, digest={}, seed={}]",
+                digest_pow(digest),
+                digest_hex(digest),
+                word_hex(seed),
+            );
+        }
+        Ok(seed)
+    }
+
+    #[cfg(feature = "std")]
+    pub fn get_account_seed_inner(
+        send: Sender<(Digest, Word)>,
+        stop: OnceLock<bool>,
+        init_seed: [u8; 32],
+        account_type: AccountType,
+        on_chain: bool,
+        code_root: Digest,
+        storage_root: Digest,
+    ) {
+        let init_seed: Vec<[u8; 8]> =
+            init_seed.chunks(8).map(|chunk| chunk.try_into().unwrap()).collect();
+        let mut current_seed: Word = [
+            Felt::from(init_seed[0]),
+            Felt::from(init_seed[1]),
+            Felt::from(init_seed[2]),
+            Felt::from(init_seed[3]),
+        ];
+        let mut current_digest = Self::compute_digest(current_seed, code_root, storage_root);
+
+        #[cfg(feature = "log")]
+        {
+            log::info!(
+                "Generating new account seed [pow={}, digest={}, seed={}]",
+                digest_pow(current_digest),
+                digest_hex(current_digest),
+                word_hex(current_seed),
+            );
+        }
+
+        let mut count = 0;
+        #[cfg(feature = "log")]
+        let mut best: (u32, Digest, Word) =
+            (digest_pow(current_digest), current_digest, current_seed);
+
+        // loop until we have a seed that satisfies the specified account type.
+        loop {
+            count += 1;
+
+            #[cfg(feature = "log")]
+            {
+                let current = (digest_pow(current_digest), current_digest, current_seed);
+                if best.0 <= current.0 {
+                    best = current;
+                }
+
+                if count % 500_000 == 0 {
+                    log::debug!(
+                        "Account loop, best seed [count={}, pow={}, digest={}, seed={}]",
+                        count,
+                        best.0,
+                        digest_hex(best.1),
+                        word_hex(best.2),
+                    );
+                }
+            }
+
+            // regularly check if another thread found a digest
+            if count % 500_000 == 0 && stop.get().is_some() {
+                return;
+            }
```

Comment:
I didn't read the whole implementation, but I think `stop.get()` does a syscall, so instead of checking on every iteration, this checks the flag every 10 seconds or so.